src/KubeOps.Operator/Queue/TimedEntityQueue.cs

```diff
@@ -81,15 +81,25 @@ public async IAsyncEnumerator<TEntity> GetAsyncEnumerator(CancellationToken canc
 
     public void Remove(TEntity entity)
     {
-        var name = entity.Name();
-        if (name is null)
+        var key = GetKey(entity);
+        if (key is null)
         {
             return;
         }
 
-        if (_management.Remove(name, out var task))
+        if (_management.Remove(key, out var task))
         {
             task.Cancel();
         }
     }
+
+    private string? GetKey(TEntity entity)
+    {
+        if (entity.Name() is null || entity.Namespace() is null)
+        {
+            return null;
+        }
+
+        return $"{entity.Namespace()}/{entity.Name()}";
+    }
```

Comment:
Seems reasonable, however, namespaces can be null right? When using the default namespace, they could theoretically be null? Or am I wrong?

Reply:
Updated this @buehler, please take a look